backends/ebpf/CMakeLists.txt

```diff
@@ -35,7 +35,26 @@ set (P4C_EBPF_SRCS
   ebpfModel.cpp
   midend.cpp
   lower.cpp
-  )
+  ../bmv2/common/helpers.cpp
+  ../bmv2/common/JsonObjects.cpp
+  ../bmv2/common/action.cpp
+  ../bmv2/common/controlFlowGraph.cpp
+  ../bmv2/common/deparser.cpp
+  ../bmv2/common/expression.cpp
+  ../bmv2/common/extern.cpp
+  ../bmv2/common/globals.cpp
+  ../bmv2/common/header.cpp
+  ../bmv2/common/lower.cpp
+  ../bmv2/common/metermap.cpp
+  ../bmv2/common/parser.cpp
+  ../bmv2/common/programStructure.cpp
```

Comment:
It looks to me that this is in fact the only file you need from bmv2, but you bring all the other ones because they are included. There must be a better way to do this.
I would start by moving psaSwitch.cpp outside of the bmv2 backend. That should rip out all dependencies that are not needed.
Do you want to talk about it before I review this PR?

Reply:
Yes, I'm fine with everything what can make the code cleaner and maintainable.

Reply:
I'd need some guidance from you/others, because I'm not so familiar with the BMv2 part.

Reply:
I have the impression you only need PsaProgramStructure.
Can you try to move that to a separate header file and only use that?
Let me know if you have trouble doing that.

Reply:
@mbudiu-vmw I've spent some time trying to refactor it a bit. The problem is that `PsaProgramStructure` contains `createX` functions (e.g., `createParsers`, etc.), which depends on other objects like `ActionConverter`, etc. I've not found a way to re-organize code structure to minimize dependencies. If you see a better way to make PSA objects re-usable, I'll be happy to apply it to PSA/eBPF backend, once ready. 

So far, it seems to be the only way to re-use PSA objects. In fact, we based on DPDK backend that used the same approach in the past: https://github.com/p4lang/p4c/blob/b8894efd1481fe83c03d61b674a596052e509244/backends/dpdk/CMakeLists.txt#L98.

Reply:
Let me take a look at this.

Reply:
> The problem is that PsaProgramStructure contains createX functions (e.g., createParsers, etc.), which depends on other objects like ActionConverter, etc. 

@osinstom maybe we can move the common PSA code in `backends/common` to make it reusable?